src/main/java/org/truffleruby/core/VMPrimitiveNodes.java

```diff
@@ -412,8 +413,8 @@ public DynamicObject getSection(DynamicObject section) {
                 }
 
                 Object[] objects = new Object[] {
-                        makeStringNode.executeMake(key, UTF8Encoding.INSTANCE, org.truffleruby.core.rope.CodeRange.CR_UNKNOWN),
-                        makeStringNode.executeMake(stringValue, UTF8Encoding.INSTANCE, org.truffleruby.core.rope.CodeRange.CR_UNKNOWN) };
+                        makeStringNode.executeMake(key, UTF8Encoding.INSTANCE, CodeRange.CR_7BIT),
+                        makeStringNode.executeMake(stringValue, UTF8Encoding.INSTANCE, CodeRange.CR_7BIT) };
```

Comment:
Why not `encodeAscii` here?

Reply:
Bad rebase. I'll fix it. Thanks.

Reply:
Nevermind. I misread this. This node creates the String `DynamicObject` and there's a new case there that fast paths on `CR_7BIT`.

Reply:
Right. Since this is a boundary maybe createString(encodeAscii()) but out of scope of the PR (also it converts from Ruby strings to j.l.String and back).